spec/models/course/material/folder_spec.rb

```diff
@@ -32,10 +32,16 @@
       end
     end
 
-    context 'after course was created' do
+    context 'before course validation' do
       let(:course) { build(:course) }
 
-      it 'creates a root folder' do
+      it 'builds a root folder' do
+        expect(course.material_folders).to be_empty
+
+        2.times { course.valid? }
+        expect(course.material_folders.length).to eq(1)
```

Comment:
twice is for testing this, we don't want to create 2 folders when validations runs multiple times.